File: village/sketch/cat.cpp
#include <iostream>
#include <fstream>

/*************************************************************************************************/
void cat_by_line(std::istream& src) {
    std::string line;

    while (true) {
        std::istream& _ = std::getline(src, line);

        if (!_.fail()) {
            std::cout << line << std::endl;
        } else {
            break;
        }
    }
}



/*************************************************************************************************/
int main(int argc, char* argv[]) {
    if (argc > 1) {
        for (int idx = 1; idx < argc; idx++) {
            std::ifstream src(argv[idx]);

            if (src.is_open()) {
                cat_by_line(src);
                src.close();
            }
        }
    } else {
        cat_by_line(std::cin);
    }

    return 0;
}

File: village/sketch/Cat.java
import java.lang.Exception;

import java.util.Scanner;

import java.io.InputStream;
import java.io.FileInputStream;

public class Cat { 
    public static void cat_by_line(InputStream src) {
        try {
            Scanner s = new Scanner(src);

            while (s.hasNextLine()) {
                String line = s.nextLine();
                System.out.println(line);
            }

            s.close();
        } catch (Exception e) {}
    }

    public static void main(String argv[]) {
        if (argv.length > 0) {
            for (int idx = 0; idx < argv.length; idx++) {
                try {
                    FileInputStream src = new FileInputStream(argv[idx]);

                    cat_by_line(src);
                    src.close();
                } catch (Exception e) {
                    System.out.println(e);
                }
            }
        } else {
            cat_by_line(System.in);
        }
    }
}
